ios/PresentationLayer/Tests/PresentationLayerTests/ViewModels/Onboarding/LoginViewModelTests.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  Created by Petr Chmelar on 07.03.2022
+//  Copyright © 2022 Matee. All rights reserved.
+//
+
+import DomainLayer
+import DomainStubs
+@testable import PresentationLayer
+import Resolver
+import SwiftyMocky
+import UseCaseMocks
+import XCTest
+
+class LoginViewModelTests: BaseTestCase {
+    
+    // MARK: Dependencies
+    
+    private let flowController = FlowControllerMock(navigationController: UINavigationController())
+    
+    private let loginUseCase = LoginUseCaseMock()
+    private let trackAnalyticsEventUseCase = TrackAnalyticsEventUseCaseMock()
+    
+    override func setupDependencies() {
+        super.setupDependencies()
+        
+        Resolver.register { self.loginUseCase as LoginUseCase }
+        Resolver.register { self.trackAnalyticsEventUseCase as TrackAnalyticsEventUseCase }
+        
+        Given(loginUseCase, .execute(
+            .value(.stubInvalidPassword),
+            willReturn: .error(RepositoryError(statusCode: StatusCode.httpUnathorized, message: ""))
+        ))
+        Given(loginUseCase, .execute(.any, willReturn: .just(())))
+    }
```

Comment:
I would move it into BaseTestCase (or better place) and maybe we could create our view model tests with init? Or we could find some way how to register all use cases in the one file for each test file. Because in case to using more use cases it will be bigger and I would say that we don't want this staff in each view model tests. Imagine home in Stravenky, there is so much methods and use cases (services) so then, this setup would be really big.

Reply:
That's a good point! Unfortunately, it's not that easy to move this setup somewhere else, because it is related to the tested ViewModel. 😔 For example, here we are setting what `loginUseCase` should return via `Given` from [SwiftyMocky](https://github.com/MakeAWishFoundation/SwiftyMocky). In another ViewModel we may need another values. Also we need references to these UseCases to test them later via `Verify`. I will try to look into this more, but mainly we should keep ViewModels small to avoid problems, that should now be easier with composition in SwiftUI. 🙂